.github/workflows/rust.yml

```diff
@@ -238,3 +238,36 @@ jobs:
           # Update Cargo.lock to minimal version dependencies.
           cargo update -Z minimal-versions
           cargo hack check --all-features --ignore-private
+
+  miri:
+    if: github.event_name == 'push' || github.event_name == 'schedule' ||
+      github.event.pull_request.head.repo.full_name != github.repository
+
+    name: Miri test
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
+          components: miri
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@v2
+        with:
+          tool: nextest
+
+      - name: Cache Cargo registry
+        uses: Swatinem/rust-cache@v2
+
+      - name: Miri setup
+        run: cargo miri setup
+
+      - name: Miri test
+        run: cargo miri nextest run -E 'not (test(db) | test(password_hash))' --no-fail-fast --all-features
```

Comment:
Why are we handling this differently than other tests? We could just add `#![cfg_attr(miri, ignore)]` at the top of `db.rs` and annotate the `password_hash*` tests with `#[cfg_attr(miri, ignore)]`. This will make it easier for people wanting to run Miri tests locally instead of using the CI.

As about why `password_hash` tests do not work on Miri, my guess is they're just super slow. They can easily take 500ms+ on a normal debug build without Miri, so with Miri it can be easily magnified by 100x or more. There's not much value in running these tests anyway, as this really just calls external code and does little above that.